gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopSyncableOutputStream.java

```diff
@@ -129,6 +129,17 @@
   // StorageResourceId.UNKNOWN_GENERATION_ID if unknown.
   private long curDestGenerationId;
 
+  // Whether perform a hsync() when calling hflush().
+  // When false, hflush() will be no-op.
+  // When true, hflush() simply calls hsync().
+  private boolean syncOnFlush;
+
+  // Minimum time interval (ns) between consecutive sync/hsync/hflush calls.
+  private long minSyncTimeIntervalNs;
+
+  // The timestamp where last time a sync is performed.
+  private long lastSyncTimestamp;
```

Comment:
Instead of manually tracking sync time use Guava's [RateLimiter](https://guava.dev/releases/28.2-jre/api/docs/com/google/common/util/concurrent/RateLimiter.html).